commands/qemu-build/buildimage.go

```diff
@@ -0,0 +1,127 @@
+package qemubuild
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-worker/commands/qemu-run"
+	"github.com/taskcluster/taskcluster-worker/engines/qemu/image"
+	"github.com/taskcluster/taskcluster-worker/engines/qemu/network"
+	"github.com/taskcluster/taskcluster-worker/engines/qemu/vm"
+)
+
+func buildImage(
+	log *logrus.Entry,
+	inputFile, outputFile string,
+	fromImage, novnc bool,
+	boot,
+	cdrom string,
+	size int,
+) error {
+	// Find absolute outputFile
+	outputFile, err := filepath.Abs(outputFile)
+	if err != nil {
+		log.Error("Failed to resolve output file, error: ", err)
+		return err
+	}
+
+	// Create temp folder for the image
+	tempFolder, err := ioutil.TempDir("", "taskcluster-worker-build-image-")
+	if err != nil {
+		log.Error("Failed to create temporary folder, error: ", err)
+		return err
+	}
+	defer os.RemoveAll(tempFolder)
+
+	var img *image.MutableImage
+	if !fromImage {
+		// Read machine definition
+		machine, err2 := vm.LoadMachine(inputFile)
+		if err2 != nil {
+			log.Error("Failed to load machine file from ", inputFile, " error: ", err2)
+			return err2
+		}
+
+		// Construct MutableImage
+		log.Info("Creating MutableImage")
+		img, err2 = image.NewMutableImage(tempFolder, int(size), machine)
+		if err2 != nil {
+			log.Error("Failed to create image, error: ", err2)
+			return err2
+		}
+	} else {
+		img, err = image.NewMutableImageFromFile(inputFile, tempFolder)
+		if err != nil {
+			log.Error("Failed to load image, error: ", err)
+			return err
+		}
+	}
+
+	// Create temp folder for sockets
+	socketFolder, err := ioutil.TempDir("", "taskcluster-worker-sockets-")
+	if err != nil {
+		log.Error("Failed to create temporary folder, error: ", err)
+		return err
+	}
+	defer os.RemoveAll(socketFolder)
+
+	// Setup a user-space network
+	log.Info("Creating user-space network")
+	net, err := network.NewUserNetwork(tempFolder)
+	if err != nil {
+		log.Error("Failed to create user-space network, error: ", err)
+		return err
+	}
+
+	// Create virtual machine
+	log.Info("Creating virtual machine")
+	vm := vm.NewVirtualMachine(img, net, socketFolder, boot, cdrom, log.WithField("component", "vm"))
```

Comment:
It is marginally confusing (although I guess allowed) to have the variable match the package name.